scripts/globals/combat/physical_utilities.lua

```diff
@@ -71,6 +95,253 @@ local elementalBelt = -- Ordered by element.
+    xi.zone.BALGAS_DAIS,
+    xi.zone.BEADEAUX,
+    xi.zone.QULUN_DOME,
+    xi.zone.DAVOI,
+    xi.zone.MONASTIC_CAVERN,
+    xi.zone.CASTLE_OZTROJA,
+    xi.zone.ALTAR_ROOM,
+    xi.zone.THE_BOYAHDA_TREE,
+    xi.zone.DRAGONS_AERY,
+    xi.zone.CASTLE_ZVAHL_KEEP_S,
+    xi.zone.THRONE_ROOM_S,
+    xi.zone.MIDDLE_DELKFUTTS_TOWER,
+    xi.zone.UPPER_DELKFUTTS_TOWER,
+    xi.zone.TEMPLE_OF_UGGALEPIH,
+    xi.zone.DEN_OF_RANCOR,
+    xi.zone.CASTLE_ZVAHL_BAILEYS,
+    xi.zone.CASTLE_ZVAHL_KEEP,
+    xi.zone.SACRIFICIAL_CHAMBER,
+    xi.zone.GARLAIGE_CITADEL_S,
+    xi.zone.THRONE_ROOM,
+    xi.zone.RANGUEMONT_PASS,
+    xi.zone.BOSTAUNIEUX_OUBLIETTE,
+    xi.zone.CHAMBER_OF_ORACLES,
+    xi.zone.TORAIMARAI_CANAL,
+    xi.zone.FULL_MOON_FOUNTAIN,
+    xi.zone.CRAWLERS_NEST_S,
+    xi.zone.ZERUHN_MINES,
+    xi.zone.KORROLOKA_TUNNEL,
+    xi.zone.KUFTAL_TUNNEL,
+    xi.zone.THE_ELDIEME_NECROPOLIS_S,
+    xi.zone.SEA_SERPENT_GROTTO,
+    xi.zone.VELUGANNON_PALACE,
+    xi.zone.THE_SHRINE_OF_RUAVITAU,
+    xi.zone.STELLAR_FULCRUM,
+    xi.zone.LALOFF_AMPHITHEATER,
+    xi.zone.THE_CELESTIAL_NEXUS,
+    xi.zone.WALK_OF_ECHOES,
+    xi.zone.MAQUETTE_ABDHALJS_LEGION_A,
+    xi.zone.LOWER_DELKFUTTS_TOWER,
+    xi.zone.DYNAMIS_SAN_DORIA,
+    xi.zone.DYNAMIS_BASTOK,
+    xi.zone.DYNAMIS_WINDURST,
+    xi.zone.DYNAMIS_JEUNO,
+    xi.zone.KING_RANPERRES_TOMB,
+    xi.zone.DANGRUF_WADI,
+    xi.zone.INNER_HORUTOTO_RUINS,
+    xi.zone.ORDELLES_CAVES,
+    xi.zone.OUTER_HORUTOTO_RUINS,
+    xi.zone.THE_ELDIEME_NECROPOLIS,
+    xi.zone.GUSGEN_MINES,
+    xi.zone.CRAWLERS_NEST,
+    xi.zone.MAZE_OF_SHAKHRAMI,
+    xi.zone.GARLAIGE_CITADEL,
+    xi.zone.CLOISTER_OF_GALES,
+    xi.zone.CLOISTER_OF_STORMS,
+    xi.zone.CLOISTER_OF_FROST,
+    xi.zone.FEIYIN,
+    xi.zone.IFRITS_CAULDRON,
+    xi.zone.QUBIA_ARENA,
+    xi.zone.CLOISTER_OF_FLAMES,
+    xi.zone.QUICKSAND_CAVES,
+    xi.zone.CLOISTER_OF_TREMORS,
+    xi.zone.CLOISTER_OF_TIDES,
+    xi.zone.GUSTAV_TUNNEL,
+    xi.zone.LABYRINTH_OF_ONZOZO,
+    xi.zone.ABYSSEA_ATTOHWA,
+    xi.zone.ABYSSEA_MISAREAUX,
+    xi.zone.ABYSSEA_VUNKERL,
+    xi.zone.ABYSSEA_ALTEPA,
+    xi.zone.SHIP_BOUND_FOR_SELBINA,
+    xi.zone.SHIP_BOUND_FOR_MHAURA,
+    xi.zone.PROVENANCE,
+    xi.zone.SHIP_BOUND_FOR_SELBINA_PIRATES,
+    xi.zone.SHIP_BOUND_FOR_MHAURA_PIRATES,
+    xi.zone.THRONE_ROOM_V,
+    xi.zone.HALL_OF_THE_GODS,
+    xi.zone.ABYSSEA_ULEGUERAND,
+    xi.zone.ABYSSEA_GRAUBERG,
+    xi.zone.ABYSSEA_EMPYREAL_PARADOX,
+    xi.zone.WALK_OF_ECHOES_P2,
+    xi.zone.DYNAMIS_SAN_DORIA_D,
+    xi.zone.DYNAMIS_BASTOK_D,
+    xi.zone.DYNAMIS_WINDURST_D,
+    xi.zone.DYNAMIS_JEUNO_D,
+    xi.zone.WALK_OF_ECHOES_P1
+}
+
+-- Meant to be called from WS master function, not from pDIF function.
+-- We do not want to be checking it PER HIT after all.
+xi.combat.physical.isLevelCorrectedZone = function(actor)
+    local actorZone = actor:getZoneID()
+    local result    = false
+
+    for _, pDifZone in pairs(pDifZoneList) do
```

Comment:
Look at how we use `set` in `rhapsodies.lua` to avoid having to look over this table all the time: https://github.com/LandSandBoat/server/blob/base/scripts/globals/rhapsodies.lua